### PR TITLE
 ipq40xx: Convert openmesh,a42 + openmesh,a62 to DSA + nvmem-cells

### PR DESCRIPTION
**Pull request by ecsv**
* ethernet1:
  - physical port label "Ethernet 1"
  - can be used to power the device
  - its mac address is printed on the device label
* ethernet2:
  - physical port label "Ethernet 2"

Both ports are not marked by there role (because the vendor firmware automatically detects roles) but the "Ethernet 1" port was used in the past for "WAN" functionality in OpenWrt.

As requested by @robimarko, it is also using nvmem to access the precal data + mac addresses.

### CODE REVIEW

File: target/linux/ipq40xx/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -142,6 +145,9 @@ ipq40xx_setup_macs()
 		lan_mac=$(cat /sys/firmware/mikrotik/hard_config/mac_base)
 		label_mac="$lan_mac"
 		;;
+	openmesh,a42)
+		label_mac="$(mtd_get_mac_binary "0:ART" 0x0)"
```
Comment (robimarko) [inline target/linux/ipq40xx/base-files/etc/board.d/02_network:151]: If its in binary, shouldn't NVMEM work?

That would avoid the need for preinit setting as well

Comment (ecsv) [reply target/linux/ipq40xx/base-files/etc/board.d/02_network:151]: Ok, I can have a look. But only a additional patch on top of this one. The integration of the partitions actually happens via bootargs (mtdparts) and not via devicetree. So we need now some placeholder entries in the DTS - similar to what is done for the openmesh/plasmacloud devices in ath79.

Comment (robimarko) [reply target/linux/ipq40xx/base-files/etc/board.d/02_network]: You can actually avoid defining fixed-partitions and just add a node for ART as there is now NVMEM support for dynamic partition parsers:
https://github.com/openwrt/openwrt/commit/1a9ee367343edce263f82cc91a49d796c9d45ea3

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/Documentation/devicetree/bindings/mtd/partitions?h=v6.1-rc2&id=118f3fbe517f49e17877f34fd677f7374970d92e

Comment (ecsv) [reply target/linux/ipq40xx/base-files/etc/board.d/02_network]: Thanks for the info. I have now switched the DTS to the shallow `partition-*` approach

**Comment by robimarko** [comment]
LGTM so:
`Reviewed-by: Robert Marko <robimarko@gmail.com>`

**Comment by ecsv** [comment]
Just pushed again to fix the conflicts with:

* netgear,rbr50
* netgear,rbs50
* netgear,srr60
* netgear,srs60

No changes besides the ones related to the patch context conflicts.

**Comment by ecsv** [comment]
And now rebased due github action errors regarding realtek patches which needed to be rebased - which was done already in master

**Comment by Ansuel** [comment]
@ecsv you should put the review tag on the first 2 patch instead of the last one. Can you do this change?

**Comment by ecsv** [comment]
@Ansuel : I have not added a single review tag myself. And even when I would have, why only the first two? Robert Marko <robimarko@gmail.com> gave the Reported-by to all 4.

**Comment by Ansuel** [comment]
@ecsv sorry but who added the tag ? o.o also IMHO it would be better to add the review tag just to the conversion itself and leave the use of nvmem without a review tag...

**Comment by ecsv** [comment]
> @ecsv sorry but who added the tag ? o.o also IMHO it would be better to add the review tag just to the conversion itself and leave the use of nvmem without a review tag...

@Ansuel since there is none - no one added it. And therefore, I already followed your rule to leave the conversion and have the nvmem patches without review tag.

Sorry, no idea how I remove something which is simply not there.

**Comment by Ansuel** [comment]
@ecsv oh well i'm stupid and misread reported with reviewed... anyway with this 

> LGTM so: `Reviewed-by: Robert Marko <robimarko@gmail.com>`

robi gave you the permission to add the review tag so can you pls add these tag to the first 2 patch? @robimarko or do you prefer all 4 ?

**Comment by ecsv** [comment]
> robi gave you the permission to add the review tag so can you pls add these tag to the first 2 patch? @robimarko or do you prefer all 4 ?

First you say I should not add it and now....

Give me a minute

**Comment by Ansuel** [comment]
@ecsv i'm asking robi if he want to add it to all 4... for me it should be added to just the first 2

**Comment by ecsv** [comment]
I will just leave for now. Nothing good will come out of this discussion when we continue today

**Comment by robimarko** [comment]
Its fine, my reviewed tag can be added to all 4 commits

**Comment by ecsv** [comment]
> Its fine, my reviewed tag can be added to all 4 commits

@robimarko Thanks, added to all 4 patches. Nothing else was changed

**Comment by hauke** [comment]
Thank you for the patches, I applied them to master in 66a3c32b47a9c9da155fd04045d7262b005a1de3, c6bef1b83a75d2f69a4d8904882800405df1a7b8, 2b2f14c959ad883e3859d957f5f8571e9c2ef547 and 087a8e39be56866804b09574b0d63fff7c667ad1.

**Pull request closed without merging**